src/dune_rules/pkg_rules.ml

```diff
@@ -508,8 +508,8 @@ module Action_expander = struct
         let roots = Paths.install_roots paths in
         let dir = section_dir_of_root roots section in
         Memo.return [ Value.Dir dir ]
-      | Macro (Pkg, arg) ->
-        (match String.split arg ~on:':' with
+      | Macro { macro = Pkg; payload } ->
+        (match Pform.Payload.Args.split payload with
```

Comment:
Isn't it possible to use the Macro_invocation api here?

Reply:
Yep, fixed